src/catalog/abstract_catalog.cpp

```diff
@@ -45,84 +46,95 @@ AbstractCatalog::AbstractCatalog(oid_t catalog_table_oid,
-
-  // Create catalog table
+  auto catalog_schema_name = create_plan->GetSchemaName();
+  auto catalog_database_name = create_plan->GetDatabaseName();
+  PELOTON_ASSERT(catalog_schema_name == std::string(CATALOG_SCHEMA_NAME));
+  // create catalog table
   Catalog::GetInstance()->CreateTable(
-      CATALOG_DATABASE_NAME, catalog_table_name,
+      catalog_database_name, catalog_schema_name, catalog_table_name,
       std::unique_ptr<catalog::Schema>(catalog_table_schema), txn, true);
 
-  // Get catalog table oid
+  // get catalog table oid
   auto catalog_table_object = Catalog::GetInstance()->GetTableObject(
-      CATALOG_DATABASE_NAME, catalog_table_name, txn);
+      catalog_database_name, catalog_schema_name, catalog_table_name, txn);
 
-  // Set catalog_table_
+  // set catalog_table_
   try {
     catalog_table_ = storage::StorageManager::GetInstance()->GetTableWithOid(
-        CATALOG_DATABASE_OID, catalog_table_object->GetTableOid());
+        catalog_table_object->GetDatabaseOid(),
+        catalog_table_object->GetTableOid());
+    // set database_oid
+    database_oid = catalog_table_object->GetDatabaseOid();
   } catch (CatalogException &e) {
     LOG_TRACE("Can't find table %d! Return false",
               catalog_table_object->GetTableOid());
   }
 }
 
 /*@brief   insert tuple(reord) helper function
-* @param   tuple     tuple to be inserted
-* @param   txn       TransactionContext
-* @return  Whether insertion is Successful
-*/
+ * @param   tuple     tuple to be inserted
+ * @param   txn       TransactionContext
+ * @return  Whether insertion is Successful
+ */
 bool AbstractCatalog::InsertTuple(std::unique_ptr<storage::Tuple> tuple,
                                   concurrency::TransactionContext *txn) {
   if (txn == nullptr)
     throw CatalogException("Insert tuple requires transaction");
 
-  std::unique_ptr<executor::ExecutorContext> context(
-      new executor::ExecutorContext(txn));
-  planner::InsertPlan node(catalog_table_, std::move(tuple));
-  executor::InsertExecutor executor(&node, context.get());
-  executor.Init();
-  bool status = executor.Execute();
+  std::vector<type::Value> params;
+  std::vector<std::string> columns;
+  std::vector<std::vector<std::unique_ptr<expression::AbstractExpression>>>
+      values;
+  values.push_back(
+      std::vector<std::unique_ptr<expression::AbstractExpression>>());
+  std::vector<int> result_format(tuple->GetSchema()->GetColumnCount(), 0);
+  for (size_t i = 0; i < tuple->GetSchema()->GetColumnCount(); i++) {
+    params.push_back(tuple->GetValue(i));
+    columns.push_back(tuple->GetSchema()->GetColumn(i).GetName());
+    values[0].emplace_back(
+        new expression::ConstantValueExpression(tuple->GetValue(i)));
+  }
+  auto node =
+      std::make_shared<planner::InsertPlan>(catalog_table_, &columns, &values);
 
-  return status;
+  executor::ExecutionResult this_p_status;
+  auto on_complete = [&this_p_status](
```

Comment:
We are trying to convert the old execution engine to llvm execution engine. 
Since the llvm support "insert" functionality, so we decided to change the logic. (It may not be useful in the future, cause I know there is a team working on generating pre-compiled plan for catalog)